### DISCUSSION TITLE
How to add lights to a custom vehicle, same as the stock ones

### DISCUSSION THREAD

**Discussion by MrPure3D** [carla-simulator/carla#7298] [Q&A]
Hi!

I have problems getting the lights to work on our custom Carla vehicle. I have followed the short guide in the documentation:
https://carla.readthedocs.io/en/0.9.15/tuto_content_authoring_vehicles/#lights , and added a light mesh with the correct UVs for the light mask maps. The material used is a renamed copy of a standard carla vehicle. The light Mesh is named after the blueprint with with the _lights ending. The custom vehicle also has the "has lights" checked in Vehiclefactory.
But there must be something else that needs to be assigned or added. 

Is there a better guide some where, or any tip highly appreciated?

**Comment by MrPure3D** [comment]
Has anyone an idéa? :-)

**Reply by gsmario** [reply]
Hello! Have you set the tag of that emissive mesh as this?

> Select any lights meshes in the hierarchy. Search for Tag in the Details panel and add the tag emissive.

This is necessary, otherway unreal doesn't know which mesh is the right one.

Thanks!
